src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,172 @@
+import styled from '@emotion/styled'
+import scrollIntoView from 'scroll-into-view-if-needed'
+import React, { ReactNode, useState } from 'react'
+import { color, space, radius, device } from '../../theme'
+import { ChevronLeft, ChevronRight } from '../../icons'
+import { H2, H6 } from '../Heading'
+import { Text } from '../Text'
+
+const CarouselContainer = styled.div`
+  overflow: hidden;
+`
+const Header = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  margin-block-end: ${space[8]};
+`
+const HeaderText = styled.div`
+  cursor: pointer;
+`
+const StyledH6 = styled(H6)`
+  color: ${color.action};
+  text-transform: none;
+  cursor: pointer;
+  padding-inline-start: ${space[8]};
+`
+const StyledText = styled(Text)`
+  color: ${color.foregroundMuted};
+`
+const Controls = styled.div`
+  display: flex;
+  align-items: center;
+  gap: ${space[8]};
+`
+const Control = styled.button`
+  display: none;
+
+  @media ${device.tablet} {
+    inline-size: ${space[32]};
+    block-size: ${space[32]};
+    border-radius: ${radius.md};
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    &:hover {
+      background-color: ${color.actionBackground};
+    }
+
+    &:disabled {
+      opacity: 0.4;
+      cursor: default;
+    }
+  }
+`
+interface CardProps {
+  amountOfCards: number
+  numberOfColums: number
+}
+
+const Cards = styled.div<CardProps>`
+  display: grid;
+  grid-template-columns: ${({ amountOfCards }) =>
+    `repeat(${amountOfCards}, 1fr)`};
+  width: 100%;
+  overflow-x: scroll;
+  overflow-y: hidden;
+  scrollbar-width: none;
+  grid-gap: ${space[16]};
+  scroll-behavior: smooth;
+  scroll-snap-type: x mandatory;
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
+
+  @media ${device.tablet} {
+    overflow-x: inherit;
+  }
+`
+
+interface Props {
+  title: string
+  id: string
+  subtitle?: string
+  viewAll?: string
+  onClickAll: () => void
```

Comment:
In general I don't think we should have a prop for a `view all` button or something like that, because it means that it always needs to be a button with an onClick, while you might want to pass a link instead.

Reply:
Yeah I was hesitant but checked with design and the see all is not optional - it should always be there on any carousel or feed items for that matter. And all See all will lead to a page for consistency. So that's why i thought doing it like this.

Reply:
But you also need to think from an outsiders perspective. People might use this without the see all. But even if it's not optional, that doesn't block using a ReactElement for this instead. Using a button for something that's supposed be a link is not optimal.

Reply:
Ok you are right, then I think it should be the same for title/subtitle - maybe other wants to add icons or something else on the left too